pyfr/solvers/base/inters.py

```diff
@@ -70,7 +70,8 @@ def _get_perm_for_view(self, inter, meth):
 
     def _view(self, inter, meth, vshape=()):
         vm = _get_inter_objs(inter, meth, self.elemap)
-        vm = [np.concatenate(m)[self._perm] for m in zip(*vm)]
+        vm = [np.concatenate(m) for m in zip(*vm)]
+        vm = [v if len(v) == len(inter) else v[self._perm] for v in vm]
```

Comment:
This conditional fails when p = 0, have to see what's the best way to check that

Reply:
If it makes any difference I do not think that the current shock capturing approach works at p = 0, either.